modules/caddyhttp/fileserver/browse.go

```diff
@@ -17,23 +17,42 @@ package fileserver
 import (
 	"bytes"
 	"encoding/json"
+	"fmt"
 	"html/template"
+	"io"
 	"net/http"
 	"os"
 	"path"
+	"path/filepath"
 	"strings"
 
 	"github.com/caddyserver/caddy/v2"
 	"github.com/caddyserver/caddy/v2/modules/caddyhttp"
+	"github.com/mholt/archiver/v3"
 )
 
 // Browse configures directory browsing.
 type Browse struct {
 	TemplateFile string `json:"template_file,omitempty"`
+	DirArchives   []string `json:"dir_archives,omitempty"`
 
 	template *template.Template
 }
 
+const (
+	zipContentType     = "application/zip"
+	zipExtension       = "zip"
+	tarContentType     = "application/tar"
+	tarExtension       = "tar"
+	tarGzipContentType = "application/tar+gzip"
+	tarGzipExtension   = "tar.gz"
+)
+
+var (
+	extensionToContentType = map[string]string{zipExtension: zipContentType, tarExtension: tarContentType, tarGzipExtension: tarGzipContentType}
+	contentTypeToExtension = map[string]string{zipContentType: zipExtension, tarContentType: tarExtension, tarGzipContentType: tarGzipExtension}
```

Comment:
This appears never to be used? (either that or my browser's Find In Page feature is broken)